src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -243,7 +245,10 @@ Device::Device(const Instance &inst, const VkSurfaceKHR surface, const bool pref
         // We have to use 2 different queue families
         // One for graphics and another one for presentation
         // Check which queue family index can be used for graphics
-        auto queue_candidate = VulkanSettingsDecisionMaker::find_graphics_queue_family(m_physical_device);
+        auto queue_candidate =
+            find_queue_family_index_if([&](const std::uint32_t /*index*/, const VkQueueFamilyProperties &queue_family) {
+                return (queue_family.queueFlags & VK_QUEUE_GRAPHICS_BIT) != 0u;
+            });
```

Comment:
Might be a good time to refactor this code a little now - if this code path is hit, the code won't work anyway (we need to be using the same queue for graphics and presentation).

Reply:
Good idea.

Reply:
I removed the code path which tries to setup separate queues for graphics and presentation: https://github.com/inexorgame/vulkan-renderer/pull/503/commits/49501139486a219b7dba0ec3ff8a4cd5db33769b
What other changes would you apply?

Reply:
That's all I think